core/commands/config.go

```diff
@@ -222,54 +221,49 @@ NOTE: For security reasons, this command will omit your private key and remote s
 }
 
 // Scrubs value and returns error if missing
-func scrubValue(m map[string]interface{}, key []string) error {
-	return scrub(m, key, false)
+func scrubValue(m map[string]interface{}, key []string) (map[string]interface{}, error) {
+	return scrub_map(m, key, false)
 }
 
 // Scrubs value and returns no error if missing
-func scrubOptionalValue(m map[string]interface{}, key []string) error {
-	return scrub(m, key, true)
+func scrubOptionalValue(m map[string]interface{}, key []string) (map[string]interface{}, error) {
+	return scrub_map(m, key, true)
 }
 
-func scrub(m map[string]interface{}, key []string, okIfMissing bool) error {
-	find := func(m map[string]interface{}, k string) (string, interface{}, bool) {
-		lckey := strings.ToLower(k)
-		for mkey, val := range m {
-			lcmkey := strings.ToLower(mkey)
-			if lckey == lcmkey {
-				return mkey, val, true
-			}
-		}
-		return "", nil, false
+func scrub_either(u interface{}, key []string, okIfMissing bool) (interface{}, error) {
+	if m, ok := u.(map[string]interface{}); ok {
+		return scrub_map(m, key, okIfMissing)
+	} else {
+		return scrub_value(m, key, okIfMissing)
 	}
+}
 
-	cur := m
-	for _, k := range key[:len(key)-1] {
-		foundk, val, ok := find(cur, k)
-		if !ok && !okIfMissing {
-			return errors.New("failed to find specified key")
-		}
-
-		if foundk != k {
-			// case mismatch, calling this an error
-			return fmt.Errorf("case mismatch in config, expected %q but got %q", k, foundk)
-		}
-
-		mval, mok := val.(map[string]interface{})
-		if !mok {
-			return fmt.Errorf("%s was not a map", foundk)
-		}
-
-		cur = mval
+func scrub_value(v interface{}, key []string, okIfMissing bool) (interface{}, error) {
+	if v == nil && !okIfMissing {
+		return nil, errors.New("failed to find specified key")
 	}
+	return nil, nil
+}
 
-	todel, _, ok := find(cur, key[len(key)-1])
-	if !ok && !okIfMissing {
-		return fmt.Errorf("%s, not found", strings.Join(key, "."))
+func scrub_map(m map[string]interface{}, key []string, okIfMissing bool) (map[string]interface{}, error) {
+	if len(key) == 0 {
+		return nil, nil // delete value
```

Comment:
This can cause a null reference exception if the caller expects the map to exist (which we do since we return `&cfg`). We may want to add a Go test here for this